File: javaSpringShip/src/main/java/pl/battleships/javaspringship/history/SimpleHistoryProvider.java
package pl.battleships.javaspringship.history;


import pl.battleships.core.api.HistoryProvider;
import pl.battleships.core.model.Board;
import pl.battleships.core.model.Position;

import java.util.*;

public class SimpleHistoryProvider implements HistoryProvider {

    private final Map<String, List<Position>> shots = new HashMap<>();
    private Map<String, Board> boards = new HashMap<>();

    @Override
    public Long addGame(String gameId, Board board) {
        boards.put(gameId, board);
        return 0L;
    }

    @Override
    public Long opponentShotForGame(String gameId, Position position) {
        shots.computeIfAbsent(gameId, s -> new ArrayList<>());
        shots.get(gameId).add(position);
        return 0L;
    }

    @Override
    public List<pl.battleships.core.model.Position> getAllShots(String gameId) {
        return Optional.ofNullable(shots.get(gameId)).orElse(Collections.emptyList());
    }
}

File: kotlinSpringShip/src/main/kotlin/pl/battleships/kotlinspringship/history/SimpleHistoryProvider.kt
package pl.battleships.kotlinspringship.history

import pl.battleships.core.api.HistoryProvider
import pl.battleships.core.model.Board
import pl.battleships.core.model.Position

class SimpleHistoryProvider : HistoryProvider {
    private val games: HashMap<String, Board> = hashMapOf()
    private val shots: HashMap<String, MutableList<Position>> = hashMapOf()

    override fun addGame(gameId: String, board: Board): Long {
        games.put(gameId, board)
        return 0
    }

    override fun opponentShotForGame(gameId: String, position: Position): Long {
        shots.computeIfAbsent(gameId) { it -> mutableListOf<Position>() }
        shots.get(gameId)?.add(position)
        return 0
    }

    override fun getAllShots(gameId: String): MutableList<Position> {
        return shots.get(gameId).orEmpty().toMutableList()
    }
}